R/exploratoryfactoranalysis.R

```diff
@@ -312,9 +312,23 @@ ExploratoryFactorAnalysis <- function(jaspResults, dataset, options, ...) {
 
   eigtab <- createJaspTable(gettext("Factor Characteristics"))
   eigtab$addColumnInfo(name = "comp", title = "",                         type = "string")
-  eigtab$addColumnInfo(name = "eigv", title = gettext("SumSq. Loadings"), type = "number", format = "sf:4;dp:3")
-  eigtab$addColumnInfo(name = "prop", title = gettext("Proportion var."), type = "number", format = "sf:4;dp:3")
-  eigtab$addColumnInfo(name = "cump", title = gettext("Cumulative"),      type = "number", format = "sf:4;dp:3")
+
+  # if a rotation is used, the table needs more columns
+  rotate <- options[[if (options[["rotationMethod"]] == "orthogonal") "orthogonalSelector" else "obliqueSelector"]]
+  if (rotate != "none") {
+    overTitleA <- gettext("Unrotated solution")
+    overTitleB <- gettext("Rotated solution")
+    eigtab$addColumnInfo(name = "sslU", title = gettext("SumSq. Loadings"),  type = "number", format = "sf:4;dp:3", overtitle = overTitleA)
+    eigtab$addColumnInfo(name = "propU", title = gettext("Proportion var."), type = "number", format = "sf:4;dp:3", overtitle = overTitleA)
+    eigtab$addColumnInfo(name = "cumpU", title = gettext("Cumulative"),      type = "number", format = "sf:4;dp:3", overtitle = overTitleA)
+    eigtab$addColumnInfo(name = "sslR", title = gettext("SumSq. Loadings"),  type = "number", format = "sf:4;dp:3", overtitle = overTitleB)
+    eigtab$addColumnInfo(name = "propR", title = gettext("Proportion var."), type = "number", format = "sf:4;dp:3", overtitle = overTitleB)
+    eigtab$addColumnInfo(name = "cumpR", title = gettext("Cumulative"),      type = "number", format = "sf:4;dp:3", overtitle = overTitleB)
+  } else {
+    eigtab$addColumnInfo(name = "sslU", title = gettext("SumSq. Loadings"),  type = "number", format = "sf:4;dp:3")
+    eigtab$addColumnInfo(name = "propU", title = gettext("Proportion var."), type = "number", format = "sf:4;dp:3")
+    eigtab$addColumnInfo(name = "cumpU", title = gettext("Cumulative"),      type = "number", format = "sf:4;dp:3")
```

Comment:
please remove `format = "sf:4;dp:3"`, as this is already implied by `type == "number"`.